### ISSUE TITLE
DatePicker: yearNavigator is no longer rendered as a dropdown

### ISSUE THREAD

**Issue by HSnicolascerrejon** [primefaces/primefaces#11951]
### Describe the bug

Hi!

When setting `yearNavigator="true"` the dropdown is not appearing, instead a spinner is shown. I don't know if the component has been changed deliberately or if it is a bug. Anyway, an option to choose which to use would be loved.
Users report that now they need to do lot of clicks to choose a year, due to the fact that the spinner has some "problems" entering the year manually (I think it's due to some regex applied to the input to let only certain years)
- Arrow keys doesn't work
- Selecting year and then starting to write a new year doesn't work

Also, manual input of the date is complicated to do because getting the field to have focus is very hard. When the field get focus, date popup shows, you cannot write manually because focus is lost. Then, if you click in the input, you still can't write on the field, I think it tries to show the popup again (after some test, with 'esc' you can manage to have the cursor in the input to start writing).
It would be great if, after clicking the field when the popup is oppened, it would be closed and the field would still have focus.

Regards!

[primefaces-test-master-dates.zip](https://github.com/primefaces/primefaces/files/15331487/primefaces-test-master-dates.zip)

### PrimeFaces edition

None

### PrimeFaces version

14.0.0

### JSF implementation

Mojarra

### JSF version

-

### Java version

17

**Comment by tandraschko** [comment]
you know that p:calendar is deprecated and p:datePicker should be used?

**Comment by HSnicolascerrejon** [comment]
> you know that p:calendar is deprecated and p:datePicker should be used?

sorry, it's `p:datePicker`

**Comment by melloware** [comment]
@HSnicolascerrejon it was deliberably changed to a spinner input to prevent the dropdown from showing 10 years at a time which made no sense from a UI/UX perspective it was done for accessibility.

See: https://github.com/primefaces/primefaces/issues/8950

That being said I do see what you mean about highlighting the YEAR and not being able to type that is a bug.

As for the focus issue please see the discussion here: https://github.com/primefaces/primefaces/issues/11928

**Comment by melloware** [comment]
@HSnicolascerrejon ok I have fixed it so ArrowUp and ArrowDown now work as expected.  Also for the year input if you already have 4 digits and type a new one it will replace the value.  Unfortuntunately a limitation of `input type="number"` is it doesn't allow me to check `selectionStart` and `selectionEnd` to see what you have selected already.

The new input is an HTML5 InputNumber letting the browser do the work for `min` and `max` etc.

```xml
<input class="ui-datepicker-year" 
size="6" 
maxlength="4" 
tabindex="0" 
aria-label="Year" 
type="number" 
min="2024" 
max="2034" 
step="1" 
value="2031">
```

here is a MonkeyPatch fixing those 3 issues.

```js
if (PrimeFaces.widget.DatePicker) {
    $.prime.datePicker.prototype.onTimeInputKeyDown = function(event) {
        if (this.options.disabled) {
            event.preventDefault();
            return;
        }

        switch (event.key) {
            case 'ArrowDown':
            case 'ArrowUp':
            case 'Tab':
            case 'Delete':
            case 'Backspace':
                // allow these keys
                return true;
        }

        switch (event.key) {
            case '0':
            case '1':
            case '2':
            case '3':
            case '4':
            case '5':
            case '6':
            case '7':
            case '8':
            case '9':
                var input = event.currentTarget;
                var newValue = input.value + event.key;

                // If input is for year and already full, reset value
                if (input.maxLength === 4 && input.value.length === 4) {
                    newValue = event.key;
                    input.value = "";
                }

                // Prevent adding more characters if input is at max length
                if (input.value.length >= input.maxLength) {
                    event.preventDefault();
                    event.stopPropagation();
                    return;
                }

                // For year input, only evaluate if the input is 4 digits long
                if (input.maxLength === 4 && newValue.length < 4) {
                    return;
                }

                // Parse new value and min/max limits for comparison
                newValue = parseInt(newValue, 10);
                var inputMin = parseInt(input.min, 10);
                var inputMax = parseInt(input.max, 10);

                // Prevent input if new value is outside the min/max range
                if (isNaN(newValue) || newValue < inputMin || newValue > inputMax) {
                    event.preventDefault();
                    event.stopPropagation();
                }
                break;

            default:
                event.preventDefault();
                event.stopPropagation();
        }
    };
};
```

**Comment by HSnicolascerrejon** [comment]
Ok, thank you very much, remember to update documentation, it still says "dropdown"
